drivers/scheduler.c

```diff
@@ -294,8 +294,8 @@ scheduler_register_event(scheduler_t *s, char mode, int fd,
 	event->id       = s->uuid++;
 	event->masked   = 0;
 
-	if (!s->uuid)
-		s->uuid++;
+	if (s->uuid < 0)
+		s->uuid = 1;
```

Comment:
Instead of checking for a negative uuid, maybe we could change uuid to an unsigned int? We don't need negative values anyway and we won't roll over as often.